command/server.go

```diff
@@ -3385,63 +3412,63 @@ func (c *ServerCommand) reloadSealsLocking(ctx context.Context, core *vault.Core
 // in the seal configuration files.
 // This function returns true if the newConfig was used to re-create the Seal.Access() objects. In other words,
 // if false is returned, there were no changes done to the seals.
-func (c *ServerCommand) reloadSeals(ctx context.Context, grabStateLock bool, core *vault.Core, newConfig *server.Config) (ret bool, err error) {
-	defer func() {
-		if err != nil {
-			// We do not log here, as the error will be logged higher in the call chain
-			return
-		}
-		if ret {
-			c.logger.Info("seal configuration reloaded successfully")
-		} else {
-			c.logger.Info("seal configuration was not reloaded")
-		}
-	}()
-
+func (c *ServerCommand) reloadSeals(ctx context.Context, grabStateLock bool, core *vault.Core, newConfig *server.Config) (bool, error) {
 	if core.IsInSealMigrationMode(grabStateLock) {
+		c.logger.Debug("not reloading seal configuration since Vault is in migration mode")
 		return false, nil
 	}
 
 	currentConfig := core.GetCoreConfigInternal()
 
-	// we need to persist seal information if multiseal is being enabled
-	addEnableMultiseal := !currentConfig.IsMultisealEnabled() && newConfig.IsMultisealEnabled()
+	// We only want to reload if multiseal is currently enabled, or it is being enabled
+	if !(currentConfig.IsMultisealEnabled() || newConfig.IsMultisealEnabled()) {
```

Comment:
Does this have to be in this function, or could we move this check so that it doesn't have to affect the sighup behavior?

Reply:
At least for now, we have decided that it is OK for SIGHUP not to reload seal configuration when enable_multiseal is false.